src/components/common/AppBreadcrumbs.tsx

```diff
@@ -0,0 +1,217 @@
+  branchName?: string;
+  buildId?: string;
+  buildHash?: string;
+  taskId?: string;
+  taskName?: string;
+}
+
+const AppBreadcrumbs = ({
+  page,
+  platform,
+  ownerName,
+  repositoryId,
+  repositoryName,
+  branchName,
+  buildId,
+  buildHash,
+  taskId,
+  taskName,
+  classes,
+}: Props) => {
+  const owner = (active = false) => (
+    <Crumb
+      key="owner"
+      active={active}
+      name={ownerName}
+      href={absoluteLink(platform, ownerName)}
+      Icon={GitHubIcon}
+      iconStyle={{ fontSize: 17 }}
+    />
+  );
+
+  const repository = (active = false) => (
+    <Crumb
+      key="repository"
+      active={active}
+      name={repositoryName}
+      href={absoluteLink(platform, ownerName, repositoryName)}
+      Icon={RepositoryIcon}
+      iconStyle={{ fontSize: 15.5 }}
+    />
+  );
+
+  const branch = (active = false) => (
+    <Crumb
+      key="branch"
+      active={active}
+      name={branchName}
+      href={absoluteLink(platform, ownerName, repositoryName, branchName)}
+      Icon={CallSplitIcon}
+      iconStyle={{ fontSize: 17.5 }}
+    />
+  );
+
+  const build = (active = false) => (
+    <Crumb
+      key="build"
+      active={active}
+      name={`Build for ${buildHash}`}
+      href={absoluteLink('build', buildId)}
+      Icon={InputIcon}
+      iconStyle={{ fontSize: 16, marginRight: 9 }}
+    />
+  );
+
+  const task = (active = false) => (
+    <Crumb
+      key="task"
+      active={active}
+      name={taskName}
+      href={absoluteLink('task', taskId)}
+      Icon={BookmarkBorderIcon}
+      iconStyle={{ fontSize: 18, marginLeft: -3, marginRight: 6 }}
+    />
+  );
+
+  const repositoryMetrics = (active = false) => (
+    <Crumb
+      key="repositoryMetrics"
+      active={active}
+      name="Metrics"
+      href={absoluteLink('metrics', 'repository', platform, ownerName, repositoryName)}
+      Icon={TimelineIcon}
+      iconStyle={{ fontSize: 16 }}
+    />
+  );
+
+  const ownerSettings = (active = false) => (
+    <Crumb
+      key="ownerSettings"
+      active={active}
+      name="Account Settings"
+      href={absoluteLink('settings', platform, ownerName)}
+      Icon={ManageAccountsIcon}
+      iconStyle={{ fontSize: 18 }}
+    />
+  );
+
+  const repositorySettings = (active = false) => (
+    <Crumb
+      key="repositorySettings"
+      active={active}
+      name="Repository Settings"
+      href={absoluteLink('settings', 'repository', repositoryId)}
+      Icon={SettingsOutlinedIcon}
+      iconStyle={{ fontSize: 16 }}
+    />
+  );
+
+  const crumbCreators =
+    {
+      owner: [owner],
+      repository: [owner, repository],
+      branch: [owner, repository, branch],
+      build: [owner, repository, branch, build],
+      task: [owner, repository, branch, build, task],
+      repositoryMetrics: [owner, repository, repositoryMetrics],
+      repositorySettings: [owner, repository, repositorySettings],
+      ownerSettings: [owner, ownerSettings],
+    }[page] || [];
```

Comment:
I would like to avoid transferring active prop from outside. What do you think about this:
`<AppBreadcrumbs
  platform={...}
  ownerName={...}
  otherCrumbs={[{ name: 'Metrics', icon: <Icon/> }]}
/>
`

Reply:
`otherCrumbs` or `extraCrumbs` works for me. 👍